clickstream-event-visualiser-ui/src/main/java/clickstream/event_visualiser_ui/ui/fragment/CSEvActionBottomSheet.kt

```diff
@@ -0,0 +1,64 @@
+package clickstream.event_visualiser_ui.ui.fragment
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.widget.TextView
+import clickstream.event_visualiser_ui.CSEventVisualiserUI
+import clickstream.event_visualiser_ui.databinding.FragmentCsEvActionBottomsheetBinding
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import kotlinx.coroutines.*
+
+internal class CSEvActionBottomSheet(context: Context) : BottomSheetDialog(context) {
+
+    private var job: CompletableJob = SupervisorJob()
+    private var coroutineScope = CoroutineScope(job)
+
+    private var binding: FragmentCsEvActionBottomsheetBinding =
+        FragmentCsEvActionBottomsheetBinding.inflate(
+            LayoutInflater.from(context)
+        )
+
+    init {
+        setContentView(binding.root)
+    }
+
+    var actionListener: CSEvActionCallbacks? = null
+        set(value) {
+            field = value
+            field?.run {
+                setUpClicks(binding.tvCsEvStartCapture, ::startCapture)
+                setUpClicks(binding.tvCsEvStopCapture, ::stopCapture)
+                setUpClicks(binding.tvCsEvClearData, ::clearData)
+                setUpClicks(binding.tvCsEvClose, ::close)
+            }
+        }
+
+    private fun setUpClicks(textView: TextView?, onClickAction: suspend () -> Unit) {
+        textView?.setOnClickListener {
+            coroutineScope.launch {
+                onClickAction()
+                dismiss()
+            }
+        }
+    }
+
+    override fun onAttachedToWindow() {
+        super.onAttachedToWindow()
+        job = SupervisorJob()
+        coroutineScope = CoroutineScope(job)
+    }
+
+    override fun onDetachedFromWindow() {
+        super.onDetachedFromWindow()
+        coroutineScope.cancel()
```

Comment:
we should set `coroutineScope` and `job` to null

Reply:
When bottom sheet is destroyed, wouldn't this be garbage collected ?